src/main/java/de/hs_mannheim/informatik/ct/persistence/repositories/RoomVisitRepository.java

```diff
@@ -21,53 +21,76 @@
-	 * @return All visitors that haven't checked out yet.
-	 */
-	@Query("SELECT visit " +
-			"FROM RoomVisit visit " +
-			"WHERE visit.endDate is null")
-	List<RoomVisit> findNotCheckedOutVisits();
+    @Query("SELECT visit " +
+            "FROM RoomVisit visit " +
+            "WHERE visit.room = :room AND visit.endDate is null")
+    List<RoomVisit> findNotCheckedOutVisits(@NonNull Room room);
 
-	@Query("SELECT visit " +
-			"FROM RoomVisit visit " +
-			"WHERE visit.room = :room AND visit.endDate is null")
-	List<RoomVisit> findNotCheckedOutVisits(@NonNull Room room);
+    void deleteByEndDateBefore(Date endDate);
 
-	void deleteByEndDateBefore(Date endDate);
+    @Query("SELECT NEW de.hs_mannheim.informatik.ct.model.Contact(visitTarget, visitOther) " +
+            "FROM RoomVisit visitTarget," +
+            "RoomVisit visitOther " +
+            "WHERE visitTarget.visitor = :visitor AND " +
+            "visitTarget.visitor != visitOther.visitor AND " +
+            "visitTarget.room = visitOther.room AND " +
+            "visitTarget.startDate <= visitOther.endDate AND " +
+            "visitOther.startDate <= visitTarget.endDate " +
+            "ORDER BY visitTarget.startDate")
+    List<Contact<RoomVisit>> findVisitsWithContact(@Param(value = "visitor") Visitor visitor);
 
-	@Query("SELECT NEW de.hs_mannheim.informatik.ct.model.Contact(visitTarget, visitOther) " +
-			"FROM RoomVisit visitTarget," +
-			"RoomVisit visitOther " +
-			"WHERE visitTarget.visitor = :visitor AND " +
-			"visitTarget.visitor != visitOther.visitor AND " +
-			"visitTarget.room = visitOther.room AND " +
-			"visitTarget.startDate <= visitOther.endDate AND " +
-			"visitOther.startDate <= visitTarget.endDate " +
-			"ORDER BY visitTarget.startDate")
-	List<Contact<RoomVisit>> findVisitsWithContact(@Param(value = "visitor") Visitor visitor);
+    /**
+     * Gets the total number of people currently checked in in a study room
+     *
+     * @param studyRooms room names of all study rooms
+     * @return total number of people currently checked in in a study room
+     */
+    @Query("SELECT COUNT (*) " +
+            "FROM RoomVisit rv " +
+            "WHERE rv.room.name IN :studyRooms " +
+            "AND rv.endDate is null")
+    int getTotalStudyRoomsVisitorCount(@Param("studyRooms") String[] studyRooms);
 
+    /**
+     * Gets all study rooms
+     *
+     * @param studyRooms room names of all study rooms
+     * @return List of study rooms with room infos (roomName, buildingName, maxCapacity) and current visitor count
+     */
+    @Query("SELECT NEW de.hs_mannheim.informatik.ct.model.StudyRoom" +
+            "(r.name, r.buildingName, r.maxCapacity, COUNT (rv.room.name) AS visitorCount) " +
+            "FROM Room r " +
+            "LEFT JOIN RoomVisit rv ON (r.name = rv.room.name) " +
+            "WHERE r.name IN :studyRooms " +
+            "AND rv.endDate IS null " +
+            "GROUP BY  r.name, r.buildingName, r.maxCapacity")
```

Comment:
`GROUP BY r.name` sollte reichen. Kommen so nicht alle Räume eines Gebäudes/Mit der gleichen Kapazität in eine Gruppe? Würde ich auf jeden Fall einen Test zu schreiben.

Reply:
Ja da hab ich einen Fehler gemacht, das maxCapacity gehört da auf jeden Fall weg, Danke dir.
So  wie ich Group By verstehe, würde ich aber das r.buildingName drin lassen:
Group by nimmt sich den ersten Parameter (name) und zählt dann die Reihen mit dem selben Namen. Wenn von diesen Reihen dann noch der buildingName unterschiedlich ist, werden die nochmal seperat gezählt. In dem Fall macht das aber keinen Unterschied, da ja alle Räume im selben Gebäude sind. Dachte nur falls es den gleichen Raumnamen in einem anderen Gebäude gibt müssen wir da unterscheiden.

Reply:
Wenn aber im raumname der erste Gebäudebuchstabe drin ist, dann sind die Raumnamen eigentlich komplett eindeutig oder? Dann könnte man den Gebäudenamen auch rausnehmen oder gibt es einen Fall wo der raumname allein nicht eindeutig ist?

Reply:
Jep, ist momentan eindeutig da der Raumname die ID in der Tabelle ist. Also man kann keine zwei Räume mit dem gleichen Name einfügen. Ob das immer Sinn macht ist eine andere Frage. Würde die query mit und ohne building akzeptieren, ändert ja erstmal nichts.